### PR TITLE
Document manual testing steps for QA

### LINKED ISSUE #53: JetBrains: document manual testing steps
The goal of this issue is to create a document in the sourcegraph/jetbrains repository that documents all steps to manually test the plugin's functionality. We are investing in automated tests but will continue to rely on manual testing for some features that are difficult to test automatically. These testing steps will be useful when we cut releases. We are also in the process of hiring an external firm to do manual QA and this document will be important for them to know what steps to make.

### PR DESCRIPTION
**Pull request by exigow**
Fixes https://github.com/sourcegraph/jetbrains/issues/53

### CODE REVIEW

**Comment by exigow** [comment]
@olafurpg, please take a look at this **draft** to verify if it's going in the right direction: https://github.com/sourcegraph/jetbrains/blob/exigow/testing-instructions/TESTING.md

Maybe there any spaces that I should cover more? Document has form of a checklist to make it easier for the tester to follow the scenarios.

I've described the main features and the bugs we've encountered recently. What remains is to choose the cadence dimension (one-off vs weekly vs bi-weekly vs monthly) and the OS dimension - in this case, it seems to me that checking Windows for Onboarding should be enough. The rest of the features seem to be OS-agnostic.

**Comment by olafurpg** [comment]
@exigow that looks fantastic! Nice work

**Comment by exigow** [comment]
Additional context: https://docs.google.com/document/d/18f0LVWP8HhYB1rJtp-H6FMA6gvXof9QlCb0PWJCIeaY

**Comment by exigow** [comment]
@sourcegraph/jetbrains-plugin, please take a look :eyes:. This are testing steps required for external QA. Motivation is described in linked issue (https://github.com/sourcegraph/jetbrains/issues/53).

I assume that if a certain functionality changes, then along with changes in the code, this document will also be updated. For this reason, we should care not to go into too much detail and avoid inflating this docs if some features are very similar / not so crucial / covered by other steps.

If you see something that is definitely missing, especially topics that are potentially difficult to cover by automatic tests, please point that out.

File: TESTING.md [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+
+## Onboarding
+
+### Sign in with GitHub
+
+Prerequisite: You have to **sign out** from all existing accounts.
+
+1. Navigate to `Cody` toolbar and use `Sign in with GitHub`.
+2. Browser is launched automatically and IDE freezes with spinning `Login to Sourcegraph` dialog.
+3. Authorize with valid account.
```
Comment (mkondratek) [inline TESTING.md]: nit: `a valid token`

File: TESTING.md [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+
+#### Expected behaviour
+
+* IDE should receive valid token automatically.
```
Comment (mkondratek) [inline TESTING.md]: nit: `a valid token`

File: TESTING.md [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+
+### Sign in with Google
+
+Onboarding through GitLab is similar to [Sign in with GitHub](#sign-in-with-github), and the authorization is also done through the browser. Expected behaviour is identical.
+
+## Autocomplete
+
+### Single-line autocomplete
+
+1. Paste following Java code:
```
Comment (mkondratek) [inline TESTING.md]: nit: `the following`

Comment (mkondratek) [reply TESTING.md]: the same for the following points 😉

File: TESTING.md [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+    ```java
+    // print Hello World!
+    System.out.
+    ```
+2. Place cursor at the end of the `System.out.` line.
```
Comment (mkondratek) [inline TESTING.md]: nit: `the cursor`

File: TESTING.md [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+1. Paste following Java code:
+    ```java
+    System.out.println("Hello, Cody!");
+    ```
+2. Select line and use `Cody | Commands | Explain Selected Code (Detailed)`.
+
+#### Expected behaviour
+
+* User is automatically switched to `Chat` tab.
+* Chat responds with a **high-level** and short description of the code without going into technical details.
```
Comment (mkondratek) [inline TESTING.md]: nit: `a short description`

**Review by mkondratek — APPROVED** [review]
LGTM. you could add a few articles here and there 😄 great work - finally, I learnt how to achieve  the `Cycle through autocomplete` case 😅

**Comment by exigow** [comment]
Thanks!

**Pull request merged**